packages/core/src/addresses.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  ChugSplashRegistryABI,
+  getOwnerAddress,
+  ManagedServiceArtifact,
+  EXECUTION_LOCK_TIME,
+  EXECUTOR_PAYMENT_PERCENTAGE,
+  DEFAULT_UPDATER_ADDRESS,
+  PROTOCOL_PAYMENT_PERCENTAGE,
+  DETERMINISTIC_DEPLOYMENT_PROXY_ADDRESS,
+  OZ_UUPS_UPDATER_ADDRESS,
+  ChugSplashManagerABI,
+  DEFAULT_GAS_PRICE_CALCULATOR_ADDRESS,
+  DEFAULT_CREATE2_ADDRESS,
+  OWNER_BOND_AMOUNT,
+  ChugSplashRegistryArtifact,
+  ChugSplashManagerArtifact,
+  DefaultAdapterArtifact,
+  OZUUPSOwnableAdapterArtifact,
+  OZUUPSAccessControlAdapterArtifact,
+  DefaultUpdaterArtifact,
+  OZUUPSUpdaterArtifact,
+  OZTransparentAdapterArtifact,
+} from '@chugsplash/contracts'
+import { constants, utils } from 'ethers'
+
+import { CURRENT_CHUGSPLASH_MANAGER_VERSION } from './constants'
+
+const chugsplashRegistrySourceName = ChugSplashRegistryArtifact.sourceName
+const chugsplashManagerSourceName = ChugSplashManagerArtifact.sourceName
+const defaultAdapterSourceName = DefaultAdapterArtifact.sourceName
+const OZUUPSOwnableAdapterSourceName = OZUUPSOwnableAdapterArtifact.sourceName
+const OZUUPSAccessControlAdapterSourceName =
+  OZUUPSAccessControlAdapterArtifact.sourceName
+const defaultUpdaterSourceName = DefaultUpdaterArtifact.sourceName
+const OZUUPSUpdaterSourceName = OZUUPSUpdaterArtifact.sourceName
+const OZTransparentAdapterSourceName = OZTransparentAdapterArtifact.sourceName
+
+export const getRegistryConstructorValues = () => [getOwnerAddress()]
+
+const [registryConstructorFragment] = ChugSplashRegistryABI.filter(
+  (fragment) => fragment.type === 'constructor'
+)
+const registryConstructorArgTypes = registryConstructorFragment.inputs.map(
+  (input) => input.type
+)
+
+export const getChugSplashRegistryAddress = () =>
+  utils.getCreate2Address(
+    DETERMINISTIC_DEPLOYMENT_PROXY_ADDRESS,
+    constants.HashZero,
+    utils.solidityKeccak256(
+      ['bytes', 'bytes'],
+      [
+        ChugSplashRegistryArtifact.bytecode,
+        utils.defaultAbiCoder.encode(
+          registryConstructorArgTypes,
+          getRegistryConstructorValues()
+        ),
+      ]
+    )
+  )
```

Comment:
Refactored the address constants to be function calls since that is required to allow for a configurable owner address. Since these addresses are also no longer constants, I've moved them into their own file as well.